dh.cpp

```diff
@@ -148,11 +148,7 @@ openssl_unique_ptr<DH> make_dh (const unsigned char* prime, size_t prime_len, co
 		throw Openssl_error(ERR_get_error());
 	}
 
-	if ((dh->p = BN_bin2bn(prime, prime_len, NULL)) == NULL) {
-		throw Openssl_error(ERR_get_error());
-	}
-
-	if ((dh->g = BN_bin2bn(generator, generator_len, NULL)) == NULL) {
+	if (!DH_set0_pqg(dh.get(), BN_bin2bn(prime, prime_len, NULL), NULL, BN_bin2bn(generator, generator_len, NULL))) {
```

Comment:
Please error check `BN_bin2bn` as before.

Reply:
I have separated out the calls to `BN_bin2bn` and error check their return value as before instead of relying on `DH_set0_pqg` to check for `NULL` values.